### ISSUE TITLE
How to get key value pairs of words and its predicted label in layoutlm model instead of showing it in an image?

### ISSUE THREAD

**Issue by vishal-nayak1** [NielsRogge/Transformers-Tutorials#82]
Model I am using (LayoutLM):
Is there any way to get key value pairs from BIOES tagged prediction on single image?

**Comment by vishal-nayak1** [comment]
@sankalp-wns 
Yes, I wrote a inference script for it to extract key value pairs.I cannot share the code due to company policy but will share some ideas how you can implemented it.
1. convert BIOES to BIO tag - https://github.com/jiesutd/NCRFpp/blob/master/utils/tagSchemeConverter.py
2. convert BIO tag to original  text- https://medium.com/analytics-vidhya/bio-tagged-text-to-original-text-99b05da6664
Note: reverse tokenization is not possible with layoutlm tokenizer to get original text or word so we might need to store original text along with its tokenized version something like----> (original_word, [list of tokenized words], [list of labels for tokenized word get from model prediction])
I hope it might be helpful :-)

**Comment by jyotiyadav94** [comment]
Hi @vishal-nayak1 ,
Thank you very much for sharing your ideas. 
I am working on the use case where I have to get the questions as keys and answers as values as described in the below image . 

Image 1
![image](https://user-images.githubusercontent.com/72126242/171961741-0099b84f-941d-4fe8-9f65-65ae867740cd.png)

I have tried the ideas which you have given in the above comments. I was able to get the keys and values in this way 
Image 2
![image](https://user-images.githubusercontent.com/72126242/171961914-205035ab-44aa-4530-9ecc-88ac638576d2.png)

However do you have any ideas to print the predicted out as described in the Image 1 where all the questions are described in the columns and their respective values

**Comment by jyotiyadav94** [comment]
@vishal-nayak1 
how do you get the BIOES tagged predictions from layoutLMV3 inference?
I am using the below code - https://github.com/NielsRogge/Transformers-Tutorials/tree/master/LayoutLMv3

**Comment by vishal-nayak1** [comment]
Hi @jyotiyadav94, 
With reference to your image 1, I would suggest you can define all your questions as NER labels(as key) and its answers as text to be extracted.
In your current implementation you need to use entity linking to map questions to their corresponding answers.

**Comment by jyotiyadav94** [comment]
@vishal-nayak1 ,
Thank you.
